authorization/authorization.go

```diff
@@ -20,8 +24,113 @@ const (
 	ManageMembersScope = "manage_members"
 )
 
+// CanHaveMembers returns a boolean indicating whether the specified resource type may have member Identities
 func CanHaveMembers(resourceTypeName string) bool {
 	return resourceTypeName == IdentityResourceTypeOrganization ||
 		resourceTypeName == IdentityResourceTypeTeam ||
 		resourceTypeName == IdentityResourceTypeGroup
 }
+
+// IdentityAssociation represents an association between an Identity and either another Identity or a Resource, whether by
+// membership or by having been granted a role.  It contains metadata about the Identity's relationship with the other
+// entity, including its membership state, and any roles it may have been assigned.
+type IdentityAssociation struct {
```

Comment:
This looks like a nice relationship to have, but is there any reason why this wasn't in a different PR :) ?

Reply:
Technically yes, it could have been in a separate PR however it kind of "just happened" in the course of refactoring the organization service.

Reply:
I kind of understand why :)